source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.h

```diff
@@ -81,6 +82,44 @@ class GrpcClientImpl : public Client,
 
 using GrpcClientImplPtr = std::unique_ptr<GrpcClientImpl>;
 
+// The client cache for RawAsyncClient for google grpc so channel is not created for each request.
+class AsyncClientCache {
+public:
+  virtual ~AsyncClientCache() = default;
+
+  virtual const Grpc::RawAsyncClientSharedPtr getOrCreateAsyncClient(
+      const envoy::extensions::filters::http::ext_authz::v3::ExtAuthz& proto_config) PURE;
```

Comment:
A question just out of curiousity: what's the benefit of having an abstract class in a compilation unit containing an implementation of another interface. Do you expect adding another implementation of the abstract class?

I guess the abstract class could help with reducing build time, but in this case it should be placed in a separate header file, so that only that header could be included in e.g. `source/extensions/filters/http/ext_authz/config.h`.

Reply:
I think it's left over after refactoring